components/nofrendo/memguard.c

```diff
@@ -127,7 +128,7 @@ static void *mem_guardalloc(int alloc_size, int guard_size)
    alloc_size = (alloc_size + 3) & ~3;
 
    /* allocate memory */
-   orig = malloc(alloc_size + (guard_size * 2));
+   orig = heap_caps_malloc(alloc_size + (guard_size * 2), MALLOC_CAP_DEFAULT);
```

Comment:
Why all changes here? heaps_caps_malloc(, MALLOC_CAP_DEFAULT) iirc is the same as malloc().

Reply:
I tested this code in another project that extends this emulator and it worked. Even memory allocation was my problem and I found the solution here.